modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/filter/factory/DefaultFilterFactoryImpl.java

```diff
@@ -37,15 +36,11 @@ public ExpressionVisitor<Object> getExpressionVisitor(
 		EntityModel entityModel, ObjectDefinition objectDefinition) {
 
 		return new PredicateExpressionVisitorImpl(
-			entityModel, entityModelProvider, _fieldPredicateProviderTracker,
```

Comment:
In this case, the PredicateExpressionVisitorImpl has a clear single owner. DefaultFilterFactoryImpl should construct and manage the ServiceTrackerMap, then pass it into PredicateExpressionVisitorImpl. Rather than holding the ServiceTrackerMap statically inside PredicateExpressionVisitorImpl.

We can live with static field of ServiceTrackerMap when we have no better choice, but when it is possible to properly manage its lifecycle, it is always preferred.